src/site/content/en/blog/form-fatigue-validate-user-input/index.md

```diff
@@ -0,0 +1,308 @@
+
+For example, you can provide different email validation messages based on a
+particular invalidation reason using the `validity` object property:
+
+```js/2,5
+if (!emailInput.validity.valid) {  
+  // case 1 - value missing
+  if (emailInput.validity.valueMissing)  
+    emailInput.setCustomValidity('We need your email to complete your purchase');  
+  // case 2 - pattern mismatch
+  else if (emailInput.validity.patternMismatch)  
+    emailInput.setCustomValidity('Your email address isn\'t valid (e.g. name@gmail.com)');  
+  // ...   
+} else {  
+  emailInput.setCustomValidity('');  
+}  
+```
+
+#### Case 1: Missing required email address
+
+<figure class="w-figure">
+  <img src="image-missing-field-default.png" alt="default validation message for missing field" style="max-width: 400px;">
+</figure>
+
+<!-- TODO (mdiblasio): update image / message, label should be above -->
+
+{% Compare 'worse', 'Less helpful' %}
+Default validation message is vague.
+{% endCompare %}
+
+<figure class="w-figure">
+  <img src="image-missing-field-custom.png" alt="custom validation message for missing field" style="max-width: 400px;">
+</figure>
+
+<!-- TODO (mdiblasio): update image / message, label should be above -->
+
+{% Compare 'better', 'More helpful' %}
+Custom validation message is specific.
+{% endCompare %}
+
+#### Case 2: Invalid email address
+
+<figure class="w-figure">
+  <img src="image-invalid-field-default.png" alt="default validation message for invalid field" style="max-width: 400px;">
+</figure>
+
+<!-- TODO (mdiblasio): update image / message, label should be above -->
+
+{% Compare 'worse', 'Less helpful' %}
+Default validation message is technincal and may be difficult for some users to understand.
+{% endCompare %}
+
+<figure class="w-figure">
+  <img src="image-invalid-field-custom.png" alt="custom validation message for invalid field" style="max-width: 400px;">
+</figure>
+
+<!-- TODO (mdiblasio): update image / message, label should be above -->
+
+{% Compare 'better', 'More helpful' %}
+Custom validation message is  non-technical and easy to understand.
+{% endCompare %}
+
+
+### Show feedback in real-time
+
+<!-- <div class="w-stats">
+  <div class="w-stat">
+    <p class="w-stat__figure">46<sub class="w-stat__sub">%</sub></p>
+    <p class="w-stat__desc"><a href="https://bit.ly/msites-benchmark-uk-2018">Did not implement error messaging in real-time</a></p>
+  </div>
+</div> -->
+
+
+You should check form data as users complete each field, rather than waiting for them to submit. This makes it quicker for users to correct mistakes, and makes it less likely that invalid or insecure data will be submitted.
+
++   **Use CSS pseudo-classes** `invalid` and `valid` to apply custom styling:
+
+
+```css
+input:not(:focus):invalid {  
+  background-color: #FFD9D9;  
+}  
+input:not(:focus):valid {  
+  background-color: #D9FFD9;  
+}  
+```  
+<!-- [https://codepen.io/samdutton/pen/ZPrpmL](https://codepen.io/samdutton/pen/ZPrpmL) -->
```

Comment:
Want to add this as an embedded glitch?

Reply:
What is the syntax for embedded glitch samples?

Reply:
Looking at the post template, I think this is right:

```
<div class="glitch-embed-wrap" style="height: 420px; width: 100%;">
  <iframe
    allow="geolocation; microphone; camera; midi; encrypted-media"
    src="https://glitch.com/embed/#!/embed/fav-kitties-starter?path=src/index.js&amp;previewSize=0"
    alt="fav-kitties-starter on Glitch"
    style="height: 100%; width: 100%; border: 0;">
  </iframe>
</div>
```